spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/KinesisMessageChannelBinder.java

```diff
@@ -108,8 +108,9 @@ protected MessageProducer createConsumerEndpoint(ConsumerDestination destination
 		// need to move these properties to the appropriate properties class
 		adapter.setCheckpointMode(CheckpointMode.record);
 		adapter.setListenerMode(ListenerMode.record);
-		adapter.setStartTimeout(10000);
-		adapter.setDescribeStreamRetries(1);
+		adapter.setStartTimeout(properties.getExtension().getConnectTimeout());
+		// Let the start timeout interrupt startup instead of eating the ResourceNotFoundException
+		adapter.setDescribeStreamRetries(properties.getExtension().getConnectTimeout() + 10);
```

Comment:
That's something what isn't good: `10000 + 10` retries!
Maybe let end-user to configure this option for `KinesisMessageDrivenChannelAdapter` as well?
I'd really prefer to expose all of them, but we definitely can address that in the separate issue, e.g. #9 

But for now to avoid confuses and magic numbers let's expose `describeStreamRetries` as well!
OTOH I called the property name as `startTimeout` in the `KinesisMessageDrivenChannelAdapter`, not sure why should we rename it here into `connectTimeout` since it is really not about connection...

I don't fail channel adapter for non-ACTIVE stream, because it can be configured for several of them and others might be active and fully available.

Hope this makes sense to you.

Thanks